test/post_processor/test_dual_from_marginal_probabilities.py

```diff
@@ -0,0 +1,78 @@
+# (C) Copyright IBM 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests for the MultiQubitPOVM class."""
+
+from unittest import TestCase
+
+import numpy as np
+from numpy.random import default_rng
+from povm_toolbox.library import (
+    ClassicalShadows,
+    RandomizedProjectiveMeasurements,
+)
+from povm_toolbox.post_processor import (
+    POVMPostProcessor,
+    dual_from_marginal_probabilities,
+)
+from povm_toolbox.sampler import POVMSampler
+from qiskit.circuit.random import random_circuit
+from qiskit.primitives import StatevectorSampler
+from qiskit.quantum_info import (
+    DensityMatrix,
+    SparsePauliOp,
+    Statevector,
+)
+
+
+class TestDualFromMarginalProbabilities(TestCase):
+    """Test that we can construct optimal dual of a POVM from marginal probabilities."""
+
+    def test_not_implemented(self):
+        """Test that errors are correctly raised."""
+        povm = ClassicalShadows(n_qubit=1).definition()[(0,)]
```

Comment:
The same comment as before but here mixing CS and RPMs.
Maybe you can update the not-implemented test case in all places to directly build a non-working `BasePOVM` (product or multi depending on the case) to show that is the culprit and not a particular library POVM.